webapp/channels/src/utils/constants.tsx

```diff
@@ -913,15 +913,51 @@ export const AnnouncementBarTypes = {
 };
 
 export const AnnouncementBarMessages = {
-    EMAIL_VERIFICATION_REQUIRED: t('announcement_bar.error.email_verification_required'),
-    EMAIL_VERIFIED: t('announcement_bar.notification.email_verified'),
-    LICENSE_EXPIRED: t('announcement_bar.error.license_expired'),
-    LICENSE_EXPIRING: t('announcement_bar.error.license_expiring'),
-    LICENSE_PAST_GRACE: t('announcement_bar.error.past_grace'),
-    PREVIEW_MODE: t('announcement_bar.error.preview_mode'),
-    WEBSOCKET_PORT_ERROR: t('channel_loader.socketError'),
-    TRIAL_LICENSE_EXPIRING: t('announcement_bar.error.trial_license_expiring'),
-};
+    EMAIL_VERIFICATION_REQUIRED: 'announcement_bar.error.email_verification_required',
+    EMAIL_VERIFIED: 'announcement_bar.notification.email_verified',
+    LICENSE_EXPIRED: 'announcement_bar.error.license_expired',
+    LICENSE_EXPIRING: 'announcement_bar.error.license_expiring',
+    LICENSE_PAST_GRACE: 'announcement_bar.error.past_grace',
+    PREVIEW_MODE: 'announcement_bar.error.preview_mode',
+    WEBSOCKET_PORT_ERROR: 'channel_loader.socketError',
+    TRIAL_LICENSE_EXPIRING: 'announcement_bar.error.trial_license_expiring',
+};
+
+// These messages correspond to AnnouncementBarMessages above
```

Comment:
Ideally, these might have been combined into the object above and/or moved somewhere closer to the announcement bar component, but there's some odd typing in the code that uses `AnnouncementBarMessages`, so I decided to leave that as-is for now

Reply:
Looking at the code, this seems wrong.

The constants defined here are for identifying the announcement bar, but they are not used (and should not be used) for identifying strings.

They happen to be the same in certain scenarios (`'announcement_bar.error.license_expiring'` and `announcement_bar.error.license_expired` for example). But in those cases, they are used in the proper intl component/function, so defineMessages is not needed (for those cases).

Also... `EMAIL_VERIFICATION_REQUIRED` seems to be used only in one place in the code, and seems completely wrongly used...

Reply:
Yeah, I also tried to fix this since this code does a lot of weird stuff. For some bizarre reason, these error bars are triggered by dispatching `logError` with a malformed `ServerError` object that is missing half the fields of `model.AppError` from the server and has a bunch of extra fields which just control the announcement bar.

It's also not the first time I've tried to rewrite that actually since I tried something similar for the notification permissions bar originally